### PR TITLE
fix: KeyError caused by not loading assets

### PR DESCRIPTION
**Pull request by pseudo-su**
#### Which issue(s) does this change fix?

I think this fixes the following issues

https://github.com/aws/aws-sam-cli/issues/2968

https://github.com/aws/aws-sam-cli/issues/2849

#### Why is this change necessary?

The current `_extract_assets` method doesn't seem to load the assets correctly, resulting in `KeyError`'s later down the line when trying to extract the stack from the cloud assembly

#### How does it address the issue?

Reads assets from the manifest successfully when it's in this format

Screen Shot 2021-11-26 at 10 57 34 am

#### What side effects does this change have?

I have no idea

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Comment by jfuss** [comment]
@pseudo-su Thanks for the contribution, however this isn't something we can accept at this time. We are working in parts of the CDK beta into `develop` and have taking a little bit of a different direction. We are working through these issues still (https://github.com/aws/aws-sam-cli/pull/3492 works through some of the current gaps we have in the existing implementation that is released).

You can follow https://docs.aws.amazon.com/cdk/latest/guide/sam.html for how this currently works with the stable releases of SAM. 
The short version is:
```
cdk synth --no-staging > template.yaml
sam local invoke MyFunction12345678 --no-event
```
Replace MyFunction12345678 which your function logical Id.

If there are things you think the beta does well that the above experience doesn't, please open an issue. We are always looking for targeted feedback to make the cli better, especially as we look to expand support for different IAC: #3154
